src/super_gradients/training/models/segmentation_models/unet/unet.py

```diff
@@ -0,0 +1,229 @@
+        :param aux_out_channels: list of segmentation heads out channels, usually set as num_classes or 1 for detail
+            edge heads.
+        :param dropout: dropout probability factor.
+        :param upsample_mode: see UpsampleMode for supported options.
+        :return: nn.ModuleList
+        """
+        heads = nn.ModuleList([
+            nn.Sequential(
+                SegmentationHead(ch, hid_ch, out_ch, dropout=dropout),
+                make_upsample_module(scale_factor=scale, upsample_mode=upsample_mode, align_corners=align_corners),
+            )
+            for ch, scale, hid_ch, out_ch, use_aux in zip(in_channels_list, aux_heads_factor, aux_hidden_channels,
+                                                          aux_out_channels, use_aux_list) if use_aux
+        ])
+        return heads
+
+    def forward(self, x):
+        encoder_feats = self.encoder(x)
+        x = self.decoder(encoder_feats)
+        x = self.seg_head(x)
+        if not self.use_aux_heads:
+            return x
+        aux_feats = [aux_head(feat) for feat, aux_head in zip(encoder_feats[-len(self.aux_heads):], self.aux_heads)]
+        aux_feats.reverse()
+        return tuple([x] + aux_feats)
+
+    def _remove_auxiliary_heads(self):
+        if hasattr(self, "aux_heads"):
+            del self.aux_heads
+
+    @property
+    def backbone(self):
+        return self.encoder.backbone
+
+    def initialize_param_groups(self, lr: float, training_params: HpmStruct) -> list:
+        """
+        Custom param groups for training:
+            - Different lr for head and rest, if `multiply_head_lr` key is in `training_params`.
+        """
+        multiply_head_lr = get_param(training_params, "multiply_head_lr", 1)
+
+        multiply_lr_params, no_multiply_params = self._separate_lr_multiply_params()
+        param_groups = [{"named_params": no_multiply_params, "lr": lr, "name": "no_multiply_params"},
+                        {"named_params": multiply_lr_params, "lr": lr * multiply_head_lr, "name": "multiply_lr_params"}]
+
+        return param_groups
+
+    def update_param_groups(self, param_groups: list, lr: float, epoch: int, iter: int, training_params: HpmStruct,
+                            total_batch: int) -> list:
+        multiply_head_lr = get_param(training_params, "multiply_head_lr", 1)
+        for param_group in param_groups:
+            param_group['lr'] = lr
+            if param_group["name"] == "multiply_lr_params":
+                param_group['lr'] *= multiply_head_lr
+        return param_groups
+
+    def _separate_lr_multiply_params(self):
+        multiply_lr_params, no_multiply_params = {}, {}
+        for name, param in self.named_parameters():
+            if "backbone." in name:
+                no_multiply_params[name] = param
+            else:
+                multiply_lr_params[name] = param
+        return multiply_lr_params.items(), no_multiply_params.items()
+
+    def prep_model_for_conversion(self,
```

Comment:
`prep_model_for_coversion` has a defined signature that allows the convertor to convert it.
`    def prep_model_for_conversion(self, input_size: Union[tuple, list] = None, **kwargs):`
you can keep the append_sigmoid and append_softmax, but please add input_size

Reply:
Done [1cf7d2b](https://github.com/Deci-AI/super-gradients/pull/401/commits/1cf7d2bbf5ea0d9ba0172513d78f1e24655757e1)